sdk/storage/Azure.Storage.Files.DataLake/src/FileClient.cs

```diff
@@ -0,0 +1,1080 @@
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual Response Append(
+            Stream content,
+            long offset,
+            byte[] contentHash = default,
+            string leaseId = default,
+            CancellationToken cancellationToken = default) =>
+            AppendInternal(
+                content,
+                offset,
+                contentHash,
+                leaseId,
+                async: false,
+                cancellationToken)
+            .EnsureCompleted();
+
+        /// <summary>
+        /// The <see cref="AppendAsync"/> operation uploads data to be appended to a file.  Data can only be appended to a file.
+        /// To apply perviously uploaded data to a file, call Flush Data.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update" />.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="Stream"/> containing the content to upload.
+        /// </param>
+        /// <param name="offset">
+        /// This parameter allows the caller to upload data in parallel and control the order in which it is appended to the file.
+        /// It is required when uploading data to be appended to the file and when flushing previously uploaded data to the file.
+        /// The value must be the position where the data is to be appended. Uploaded data is not immediately flushed, or written, to the file.
+        /// To flush, the previously uploaded data must be contiguous, the position parameter must be specified and equal to the length
+        /// of the file after all data has been written, and there must not be a request entity body included with the request.
+        /// </param>
+        /// <param name="contentHash">
+        /// This hash is used to verify the integrity of the request content during transport. When this header is specified,
+        /// the storage service compares the hash of the content that has arrived with this header value. If the two hashes do not match,
+        /// the operation will fail with error code 400 (Bad Request). Note that this MD5 hash is not stored with the file. This header is
+        /// associated with the request content, and not with the stored content of the file itself.
+        /// </param>
+        /// <param name="leaseId">
+        /// Optional lease id.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> describing the state
+        /// of the updated file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
```

Comment:
You should only be using this on methods that forward into Azure.Storage.Blobs.dll or call other higher level functions.  You're putting them everywhere which invalidates our distributed tracing testing.

Reply:
Unfortunately,  the diagnostics thing doesn't support calling methods in a parent class.  Adding [FowardsClientCalls] was my work around.  Unless we can get support for this in Azure Core, I don't see another way.

Reply:
Sure, they're required when you're calling another class.  But you're not doing that here.  You're calling into DataLakeRestClient for new operations and we're not validating that the distributed tracing spans are being created properly.

Reply:
For File Append and Flush, we can remove the [FowardsClientCalls].  My point still stands for the rest of the operations, which call Path Client methods.

Reply:
When I attempt to remove [FowardsClientCalls] for FileClient.Append():

 AppendDataAsync
   Source: FileClientTests.cs line 1184
   Duration: 16 ms

  Message: 
    System.InvalidOperationException : Expected diagnostic event not fired Azure.Storage.Files.DataLake.FileClient.Append.Start 
        fired events (Azure.Storage.Files.DataLake.PathClient.AppendData.Start, Azure.Core.Pipeline.DiagnosticScope+DiagnosticActivity, Azure.Storage.Files.DataLake), (Azure.Core.Http.Request.Start, Azure.Core.HttpMessage, Azure.Core), (Azure.Core.Http.Request.Stop, Azure.Core.HttpMessage, Azure.Core), (Azure.Storage.Files.DataLake.PathClient.AppendData.Stop, , Azure.Storage.Files.DataLake) 
        You may have forgotten to set your operationId to Azure.Storage.Files.DataLake.FileClient.Append.Start in AppendAsync or applied the Azure.Core.ForwardsClientCallsAttribute to AppendAsync.
  Stack Trace: 
    DiagnosticScopeValidatingInterceptor.Intercept(IInvocation invocation) line 85
    AbstractInvocation.Proceed()
    FileClientProxy.AppendAsync(Stream content, Int64 offset, Byte[] contentHash, String leaseId, CancellationToken cancellationToken)
    FileClientTests.AppendDataAsync() line 1196
    GenericAdapter`1.GetResult() line 99
    AsyncToSyncAdapter.Await(Func`1 invoke) line 60
    TestMethodCommand.Execute(TestExecutionContext context) line 64
    <>c__DisplayClass1_0.<Execute>b__0() line 58
    BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action) line 73

This is because the operation in the generated code is Azure.Storage.Files.DataLake.PathClient.AppendData

Reply:
Yes, and you need to pass an operationId with the name of your current method (just `Append`, right?).

Reply:
I could, just for Append and Flush, but this doesn't help with the larger issue that the Diagnostic Interceptor doesn't support calling a parent class's methods.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8422